ci/setup-and-build.ps1

```diff
@@ -0,0 +1,202 @@
+        "retrieve", `
+        "tools", `
+        "schema_compiler-x86_64-win32", `
+        "$($pinned_core_sdk_version)", `
+        "$($core_sdk_dir)\tools\schema_compiler-x86_64-win32.zip" `
+    )
+    Start-Process -Wait -PassThru -NoNewWindow -FilePath "spatial" -ArgumentList @(`
+        "package", `
+        "retrieve", `
+        "schema", `
+        "standard_library", `
+        "$($pinned_core_sdk_version)", `
+        "$($core_sdk_dir)\schema\standard_library.zip" `
+    )
+    Start-Process -Wait -PassThru -NoNewWindow -FilePath "spatial" -ArgumentList @(`
+        "package", `
+        "retrieve", `
+        "worker_sdk", `
+        "c-dynamic-x86-msvc_md-win32", `
+        "$($pinned_core_sdk_version)", `
+        "$($core_sdk_dir)\worker_sdk\c-dynamic-x86-msvc_md-win32.zip" `
+    )
+
+    Start-Process -Wait -PassThru -NoNewWindow -FilePath "spatial" -ArgumentList @(`
+        "package", `
+        "retrieve", `
+        "worker_sdk", `
+        "c-dynamic-x86_64-msvc_md-win32", `
+        "$($pinned_core_sdk_version)", `
+        "$($core_sdk_dir)\worker_sdk\c-dynamic-x86_64-msvc_md-win32.zip" `
+    )
+    Start-Process -Wait -PassThru -NoNewWindow -FilePath "spatial" -ArgumentList @(`
+        "package", `
+        "retrieve", `
+        "worker_sdk", `
+        "c-dynamic-x86_64-gcc_libstdcpp-linux", `
+        "$($pinned_core_sdk_version)", `
+        "$($core_sdk_dir)\worker_sdk\c-dynamic-x86_64-gcc_libstdcpp-linux.zip" `
+    )
+    Finish-Event "download-spatial-packages" "build-unreal-gdk-:windows:"
+
+
+    Start-Event "extract-spatial-packages" "build-unreal-gdk-:windows:"
+    Expand-Archive -Path "$($core_sdk_dir)\tools\schema_compiler-x86_64-win32.zip" -DestinationPath "$($binaries_dir)\Programs\" -Force
+    Expand-Archive -Path "$($core_sdk_dir)\schema\standard_library.zip" -DestinationPath "$($binaries_dir)\Programs\schema\" -Force
+    Expand-Archive -Path "$($core_sdk_dir)\worker_sdk\c-dynamic-x86-msvc_md-win32.zip" -DestinationPath "$($binaries_dir)\Win32\" -Force
+    Expand-Archive -Path "$($core_sdk_dir)\worker_sdk\c-dynamic-x86_64-msvc_md-win32.zip" -DestinationPath "$($binaries_dir)\Win64\" -Force
+    Expand-Archive -Path "$($core_sdk_dir)\worker_sdk\c-dynamic-x86_64-gcc_libstdcpp-linux.zip" -DestinationPath "$($binaries_dir)\Linux\" -Force
+    Finish-Event "extract-spatial-packages" "build-unreal-gdk-:windows:"
+
+
+    # Copy from binaries_dir
+    Copy-Item "$($binaries_dir)\Win64\include\*" "$($worker_sdk_dir)\" -Force -Recurse
+
+    Write-Log "Fetch MSBUILD_EXE location"
+    $msbuild_exe = "${env:ProgramFiles(x86)}\MSBuild\14.0\bin\MSBuild.exe"
+
+
+    Start-Event "build-utilities" "build-unreal-gdk-:windows:"
+    #%MSBUILD_EXE% /nologo /verbosity:minimal .\SpatialGDK\Build\Programs\Improbable.Unreal.Scripts\Improbable.Unreal.Scripts.sln /property:Configuration=Release
+    $msbuild_proc = Start-Process -Wait -PassThru -NoNewWindow -FilePath "$($msbuild_exe)" -ArgumentList @(`
```

Comment:
From looking at https://buildkite.com/improbable/unrealgdk-premerge/builds/36#65d3be64-7446-4b80-8a17-19ace05046c8/200 it seems like this is doing the same thing we were having problems with in the UE build:

1. Start a quick msbuild process
1. After the build completes, one of the MS compilers stays running for 10 minutes, just in case you want to use it again soon.
1. Powershell's `Start-Process` with `-Wait` will block until the command you supply, plus all child processes _including that MS compiler that hangs around for 10 mins_ have exited, thus blocking the script for 10 mins for no good reason.

I'll take a look, but IIRC what we ended up doing to get around this was:
1. Use `Start-Process` without `-Wait`
1. Immediately call `Wait-Process -Id (Get-Process -InputObject $process_obj).id`, which will wait for _only_ the named process, and does not care about its child processes.

That way, once your actual MSBuild invocation is done, the script should immediately move on and ignore the stupid MS compiler trying to be clever.

Reply:
The same build run shows the same problem on the `RunUAT.bat` invocation below that actually builds the GDK.

If we can fix the unnecessary waiting, it'll cut the total time for this pipeline from 29 mins down to 9.

Reply:
This is fixed now